nixos/modules/services/networking/unifi.nix

```
@@ -27,24 +27,19 @@ in
       '';
     };
 
-    services.unifi.jrePackage = lib.mkOption {
-      type = lib.types.package;
-      default = if (lib.versionAtLeast (lib.getVersion cfg.unifiPackage) "7.5") then pkgs.jdk17_headless else if (lib.versionAtLeast (lib.getVersion cfg.unifiPackage) "7.3") then pkgs.jdk11 else pkgs.jre8;
-      defaultText = lib.literalExpression ''if (lib.versionAtLeast (lib.getVersion cfg.unifiPackage) "7.5") then pkgs.jdk17_headless else if (lib.versionAtLeast (lib.getVersion cfg.unifiPackage) "7.3" then pkgs.jdk11 else pkgs.jre8'';
-      description = ''
-        The JRE package to use. Check the release notes to ensure it is supported.
+    services.unifi.jrePackage = lib.mkPackageOption pkgs "jdk" {
+      default = "jdk17_headless";
+      extraDescription = ''
+        Check the UniFi controller release notes to ensure it is supported.
       '';
     };
 
-    services.unifi.unifiPackage = lib.mkPackageOption pkgs "unifi5" { };
+    services.unifi.unifiPackage = lib.mkPackageOption pkgs "unifi" {
+      default = "unifi8";
+    };
 
     services.unifi.mongodbPackage = lib.mkPackageOption pkgs "mongodb" {
-      default = "mongodb-5_0";
-      extraDescription = ''
-        ::: {.note}
-        unifi7 officially only supports mongodb up until 4.4 but works with 5.0.
-        :::
-      '';
+      default = "mongodb-7_0";
```

```suggestion
      default = "mongodb-6_0";
```

https://www.mongodb.com/docs/manual/release-notes/7.0-upgrade-standalone/#upgrade-version-path
> To upgrade an existing MongoDB deployment to 7.0, you must be running a 6.0-series release.
> 
> To upgrade from a version earlier than the 6.0-series, you must successively upgrade major releases until you have upgraded to 6.0-series. For example, if you are running a 5.0-series, you must [upgrade first to 6.0](https://www.mongodb.com/docs/manual/release-notes/6.0/#std-label-6.0-upgrade) before you can upgrade to 7.0.

---

This default will only be used for new installations. The `stateVersion` conditional below ensures that users on older versions will get an error message telling them to set the packages explicitly. Maybe the guidance could be more specific though? I don’t know if there’s even a clean upgrade path from the older UniFi controller version, though.

---

UniFi upgrades are generally fairly smooth in my experience for what it's worth; from 7 -> 8 was just change `unifiPackage` and switch. 

For MongoDB, it might be worth including the MongoDB upgrade path link and a note that stepping through intermediate versions may be required as part of the error message?

---

I had a go at expanding the message; let me know what you think.

---

Looks good to me. Thanks.